### PR TITLE
Bench for logs

### PR DESCRIPTION
**Pull request by Rockin2**
## Changes

Add benches for logs.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/benches/log.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+    group.finish();
+}
+
+fn criterion_benchmark(c: &mut Criterion) {
+    log_benchmark_group(c, "simple-log", |logger| {
+        logger.emit(LogRecord::builder().with_body("simple log".into()).build())
+    });
+
+    log_benchmark_group(c, "long-log", |logger| {
```
Comment (cijothomas) [inline opentelemetry-sdk/benches/log.rs:63]: not sure if we need to distinguish between body being short vs long. SDK is not doing any processing with Body content anyway, so this is less valuable.

**Review by cijothomas — APPROVED** [review]
Looks good to get started with benchmarking logging signal! Left couple nits, totally okay to consider for a future improvement as well.

File: opentelemetry-sdk/benches/log.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                .with_attribute("event.id", 20)
+                .with_attribute("user.name", "otel")
+                .with_attribute("user.email", "otel@opentelemetry.io")
+                .build(),
+        )
+    });
+
+    log_benchmark_group(c, "full-log-with-9-attributes", |logger| {
+        logger.emit(
+            LogRecord::builder()
```
[the remark this answers is not in the document]
Comment (Rockin2) [reply opentelemetry-sdk/benches/log.rs:98]: I could do that in another PR. :)

File: opentelemetry-sdk/benches/log.rs (hunk 1 above)
Comment (Rockin2) [reply opentelemetry-sdk/benches/log.rs:63]: Do you want me to delete this bench?

Comment (lalitb) [reply opentelemetry-sdk/benches/log.rs:63]: I believe no harm in keeping. SDK does internally copy the argument passed in `with_body`, so gives a good idea of the cost for copy operation.

File: opentelemetry-sdk/benches/log.rs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                .with_severity_text(Severity::Warn.name())
+                .with_attribute("name", "my-event-name")
+                .with_attribute("event.id", 20)
+                .with_attribute("user.name", "otel")
+                .with_attribute("user.email", "otel@opentelemetry.io")
+                .with_attribute("log.source.file.name", "log.rs")
+                .with_attribute("log.source.file.path", "opentelemetry_sdk/benches/log.rs")
+                .with_attribute("log.source.file.line", 96)
+                .with_attribute("log.module.path", "opentelemetry_sdk::benches::log")
+                .with_attribute("log.target", "opentelemetry_sdk::benches::log")
```
Comment (lalitb) [inline opentelemetry-sdk/benches/log.rs:112]: Good to have the test for all the attribute types (or at-least - bool, int, float, string, byte) - https://github.com/open-telemetry/opentelemetry-rust/blob/443f68b1b8cd9016aec7f21ce913884f473c848c/opentelemetry/src/logs/record.rs#L79-L95)

File: opentelemetry-sdk/benches/log.rs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            LogRecord::builder()
+                .with_body("full log".into())
+                .with_timestamp(now)
+                .with_observed_timestamp(now)
+                .with_severity_number(Severity::Warn)
+                .with_severity_text(Severity::Warn.name())
+                .with_attribute("name", "my-event-name")
+                .with_attribute("event.id", 20)
+                .with_attribute("user.name", "otel")
+                .with_attribute("user.email", "otel@opentelemetry.io")
```
Comment (lalitb) [inline opentelemetry-sdk/benches/log.rs:91]: Can we also add test for `with_attributes` ?

File: opentelemetry-sdk/benches/log.rs (hunk 3 above)
Comment (Rockin2) [reply opentelemetry-sdk/benches/log.rs:112]: Ok. I can have a test for each attributes type

File: opentelemetry-sdk/benches/log.rs (hunk 4 above)
Comment (Rockin2) [reply opentelemetry-sdk/benches/log.rs:91]: The thing is with this test I don't know if I will create the Voc outside of the bench or inside? What is your opinion?

Comment (lalitb) [reply opentelemetry-sdk/benches/log.rs:91]: Vec should be outside of the bench.

**Pull request merged**